openmc/mesh.py

```diff
@@ -210,3 +227,90 @@ def get_mesh_xml(self):
             subelement.text = ' '.join(map(str, self._width))
 
         return element
+
+    def build_cells(self, bc=['reflective'] * 6):
+        """Generates a list of cells which mimic the mesh geometry.
+
+        Parameters
+        ----------
+        bc : iterable of {'reflective', 'periodic', or 'vacuum'}
+            Boundary conditions for each of the six faces of a parallelopiped:
+            North, East, South, West, Up, and Down.
+
+        Returns
+        -------
+        root_cell : openmc.Cell
+            The cell containing the lattice mimicking the mesh geometry.
+        cells : iterable of openmc.Cell
+            The list of cells within each lattice position mimicking the mesh
+            geometry.
+
+        """
+
+        cv.check_length('bc', bc, length_min=6, length_max=6)
+        for entry in bc:
+            cv.check_value('bc', entry, ['transmission', 'vacuum',
+                                         'reflective', 'periodic'])
+
+        if len(self.dimension) == 2:
+            twod = True
+        else:
+            twod = False
+
+        # Build enclosing cell
+        xplanes = [openmc.XPlane(x0=self.lower_left[0],
+                                 boundary_type=bc[3]),
+                   openmc.XPlane(x0=self.upper_right[0],
+                                 boundary_type=bc[1])]
+        yplanes = [openmc.YPlane(y0=self.lower_left[1],
+                                 boundary_type=bc[2]),
+                   openmc.YPlane(y0=self.upper_right[1],
+                                 boundary_type=bc[0])]
+        if twod:
+            zplanes = [openmc.ZPlane(z0=-1.E50, boundary_type=bc[5]),
+                       openmc.ZPlane(z0=+1.E50, boundary_type=bc[5])]
+        else:
+            zplanes = [openmc.ZPlane(z0=self.lower_left[2],
+                                     boundary_type=bc[5]),
+                       openmc.ZPlane(z0=self.upper_right[2],
+                                     boundary_type=bc[5])]
+        root_cell = openmc.Cell()
+        root_cell.region = ((+xplanes[0] & -xplanes[1]) &
+                            (+yplanes[0] & -yplanes[1]) &
+                            (+zplanes[0] & -zplanes[1]))
+
+        # Build our universes
```

Comment:
Ditto - elaborate on this comment a bit to describe what each universe represents